tests/models/test_dagrun.py

```diff
@@ -564,6 +565,27 @@ def with_all_tasks_removed(dag):
         flaky_ti.refresh_from_db()
         self.assertEqual(State.NONE, flaky_ti.state)
 
+    @mock.patch('airflow.models.dagrun.task_instance_mutation_hook')
```

Comment:
Can you add a test for TI where it is UP_FOR_RETRY

Reply:
@milton0825 Can you add a test for TI where it is UP_FOR_RETRY

Reply:
Testing that probably won't add too much value as the logic is not based on the state of TI.

Reply:
https://github.com/apache/airflow/pull/8852#discussion_r428262018 Based on our discussion in that comment if in future someone removes L468, no test would fail but it would break the behaviour that ti_mutation_hook won't work on ti that are up for retry.

Reply:
added